ui/components/common/Validators.js

```diff
@@ -0,0 +1,41 @@
+const Validators = {
+    String: {
+        nonBlank: s => s
+                && typeof s === typeof ""
+                && s.length > 0,
+        defined: s => s && typeof s === typeof "",
+        minLength: l => s => Validators.String.nonBlank(s) && s.length >= l,
+        maxLength: l => s => Validators.String.nonBlank(s) && s.length < l
+    },
+    Number: {
+        natural: n => !Number.isNaN(n) && parseInt(n, 10) > 0
+    },
+    Boolean: {
+        valid: b => typeof b === typeof true
+    }
+};
+
+/**
+ * Run a number of validations that have to be passed in an array of the form:
+ * [
+ *  <Validation Function 1>, <Validation Subject 1>
+ *  ...
+ *  <Validation Function N>, <Validation Subject N>
+ * ]
+ * @param validations The validations array
+ * @returns {string} "success" on success, otherwise "error"
+ */
```

Comment:
👍 Great comment !!!

Reply:
Sure, i can add a real example

Reply:
@pb82 it was merged without a better explanation.

Reply:
I added an example of how to use the validation function. It's not really general and only used for the dialogs in the UI.